homeassistant/components/device_tracker/device_action.py

```diff
@@ -0,0 +1,67 @@
+"""Provides device automations for Device tracker."""
+from typing import List, Optional
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_DEVICE_ID,
+    CONF_DOMAIN,
+    CONF_ENTITY_ID,
+    CONF_TYPE,
+    STATE_HOME,
+    STATE_NOT_HOME,
+)
+from homeassistant.core import Context, HomeAssistant, split_entity_id
+from homeassistant.helpers import entity_registry
+import homeassistant.helpers.config_validation as cv
+
+from . import DOMAIN, SERVICE_SEE
+from .const import ATTR_LOCATION_NAME, ATTR_DEV_ID
+
+ACTION_TYPES = {"comes_home", "leaves_home"}
```

Comment:
Actually having second thoughts on this now. 

See device tracking can be about coming home/leaving home, if it's source type is router. But should we call it entering/leaving home or should it be online/offline? 

If it's GPS, we should have other action types, like entering/leaving zones. Should be in another PR.